Automattic/wp-calypso
# Editor: Remove more unused post editor code

In #46331 we removed a bunch of unused legacy post editor code. 

This PR follows up by removing even more of it that becomes obvious when we move an independent component somewhere else.

#### Changes proposed in this Pull Request

* Move `PublicizeMessage` to its own component directory in `components`.
* Remove unused `displayMessageHeading` prop from `PublicizeMessage` - we're removing the only code that actually uses that prop's default value.
* Fix `PublicizeMessage`'s class names to be compliant with our ESLint rules
* Remove editor drawer, editor sharing, editor fieldset, editor term selector, and unused related components.

#### Testing instructions

* Go to `/posts/:site` where `:site` is a public site with posts and at least one Publicize connection, click the ellipsis on the right of a post, and click "Share".
* Verify the publicize message box looks and works as it did before.
* Verify none of the code we're removing is used.
* All tests should pass
